### PR TITLE
Use numba njit code to inject NSB in R1 waveforms

### PR DESCRIPTION
**Pull request by gabemery**
New version of #1006 done in a more recent version of lstchain (the diff ended up being the same, but it cleans up the commit history). Significantly accelerate NSB injection on waveforms compared to the current implementation using scipy.interpd1d every time.

### CODE REVIEW

File: lstchain/reco/reconstructorCC.py [hunk 1]
```diff
@@ -212,18 +210,19 @@ def template_interpolation(gain, times, t0, dt, a_hg, a_lg, size):
...
     n, m = times.shape
+    size = a_hg.shape[0]
     out = np.empty((n, m))
     for i in range(n):
         for j in range(m):
             # Find the index before the requested time
             a = (times[i, j]-t0)/dt
             t = int(a)
-            if a < size:
+            if t+1 < size:
```
Comment (rlopezcoto) [inline lstchain/reco/reconstructorCC.py:220]: why was this modified in one unit?

File: lstchain/reco/reconstructorCC.py [hunk 2]
```diff
@@ -192,7 +192,7 @@ def template_interpolation(gain, times, t0, dt, a_hg, a_lg, size):
     ----------
     gain: boolean 1D array
         Gain channel used per pixel
-    times: float64 1D array
+    times: float64 2D array
```
Comment (rlopezcoto) [inline lstchain/reco/reconstructorCC.py:195]: did the variable `times` change its shape or was the description wrong before?

File: lstchain/reco/reconstructorCC.py (hunk 1 above)
Comment (maxnoe) [reply lstchain/reco/reconstructorCC.py:220]: Use <= instead of the awkward +  1 <

Comment (gabemery) [reply lstchain/reco/reconstructorCC.py:220]: Sorry I was wrong in what I was saying before since a<size also guaranties that t is less than size. The change is now quite old so I didn't fully remember. 
But we use both the t and t+1 element in the following code. So the error occurred when t+1 was equal to size. Since the array for interpolation was usually binned in fractions of nano second it was a rare bug I saw quite late. (there is no a==size issue).

I made the decision on checking on the value of t instead of a in the fix because this is what is used in the array afterwards.

The needed criteria is thus that t+1 is the index of an array element (so < size = len(array)).

@maxnoe <= wouldn't work, it goes in the wrong direction. It would be +2 <=

Comment (maxnoe) [reply lstchain/reco/reconstructorCC.py:220]: ah, right, sorry...

**Review by rlopezcoto — APPROVED** [review]

**Comment by gabemery** [comment]
I was told that a problematic behaviour was observed with this branch in an extreme injection case (nominal NSB X 32). I will try to reproduce/understand the issue. Since users are still limited for this feature I propose to wait for this check before merging.

**Comment by gabemery** [comment]
2 Issues is fixed:
 - generally important : add check that we don't interpolate before the start of the provided template when injecting pulses (0 < t+1 in reconstructorCC.py)
 - small improvement for small injection, important for larger ones : extended the pre-signal time window for pulse injection by 5 samples to better account for tails in the pulse (for information, the pulse peaks at ~9ns and previous version injected pulses starting up to 20ns before the data. Now it is 25ns. I tried 30ns without seeing effects)
